### DISCUSSION TITLE
Using openwakeword tflite with flutter

### DISCUSSION THREAD

**Discussion by hasanalaniIT** [dscripka/openWakeWord#177] [Q&A] [answered]
Hello guys I'm building my own virtual assistant and I'm facing a problem with using the wake word due to the shape of the model 

what I'm trying to do is 
to use a microphone that passes raw data byte stream every second on a sample rate of 16000 HZ and length of 16000
but the shape of the model is (1, 28, 96)  I tried to reshape my audio data but it does not work because it needs audio data length or size of (2688)
so this is the simple flow that I want to achieve :
mic data 16000 len --> openwakeword tflite model -->> result  --> repeat on no detection 
what are the steps needed to achieve this flow or behavior with openwakeword models?

**Comment by dscripka** [comment, marked as answer]
Using openWakeWord actually involves 3 separate models working together: a melspectrogram model, an embedding model, and then the wakeword prediction model. You can see how all three of these combine together [here](https://github.com/dscripka/openWakeWord/blob/c40fe924ffa12e9ddf24a3e5fcdeb4fd58ab07eb/openwakeword/utils.py#L409).

Re-implementing this in flutter is possible, but you'll have to match the logic that prepares the audio and passes everything through these three models in the proper order.

**Comment by Cheems194** [comment]
@hasanalaniIT Did you succeed in your implementation?
